tests/samplers_tests/tpe_tests/test_sampler.py

```diff
@@ -1054,6 +1056,37 @@ def test_group_experimental_warning() -> None:
         _ = TPESampler(multivariate=True, group=True)
 
 
+def run_tpe(k: int, sequence_dict: Dict[int, List[int]], hash_dict: Dict[int, int]) -> None:
+    hash_dict[k] = hash("nondeterministic hash")
+    sampler = TPESampler(n_startup_trials=1, seed=2, multivariate=True, group=True)
+    study = create_study(sampler=sampler)
+    sequence = []
+    for _ in range(10):
+        trial = study.ask()
+        picked = [i for i in range(10) if trial.suggest_int(str(i), 0, 1) == 1]
+        study.tell(trial, len(picked))
+        sequence.extend(picked)
+    sequence_dict[k] = sequence
+
+
+# Test deterministic iteration as discussed in https://github.com/optuna/optuna/issues/3137
```

Comment:
Looks like we don't have to note the issue here. I think the comments in the sampler.py are sufficient.